alghaibb/hangout-haven
# Add sidebar with navigation with dynamic routes

## Enhancement: Sidebar Navigation with Dynamic Link Styling

### Introduction
This pull request introduces a significant enhancement to the sidebar navigation component of our application. By leveraging the `usePathname` hook from Next.js, we have now implemented a feature where navigation links dynamically change their styling based on the current route. This update greatly improves user experience by visually indicating the active page, making navigation more intuitive and engaging.

### Implementation Details
- **Dynamic Active Link Styling**: The sidebar now identifies the active route using the `usePathname` hook and applies a distinct background color to the corresponding link. This is done through a combination of conditional rendering and the `cn` utility function, which dynamically assigns classNames based on the link's active state.
- **Sidebar Link Management**: Utilized the `sidebarLinks` constant from "@/constants" to manage the navigation links in a centralized manner. This approach not only simplifies the management of sidebar links but also enhances maintainability.

### How to Test
1. Navigate through the application to observe the sidebar behavior.
2. Confirm that the active link is correctly highlighted based on the current route.
3. Test responsiveness and verify that the sidebar maintains its functionality across different screen sizes.
